examples/native-fungible/src/service.rs

```diff
@@ -35,28 +36,42 @@ impl Service for NativeFungibleToken {
         _context: &QueryContext,
         request: Request,
     ) -> Result<Response, Self::Error> {
-        let schema =
-            Schema::build(QueryRoot, Operation::mutation_root(), EmptySubscription).finish();
+        let schema = Schema::build(
+            QueryRoot {
+                context: self.context().clone(),
+            },
+            Operation::mutation_root(),
+            EmptySubscription,
+        )
+        .finish();
         let response = schema.execute(request).await;
         Ok(response)
     }
 }
 
-struct QueryRoot;
+struct QueryRoot {
+    context: ViewStorageContext,
+}
 
 #[Object]
 impl QueryRoot {
-    async fn balance(&self, owner: AccountOwner) -> Result<Amount, async_graphql::Error> {
-        let owner = match owner {
-            AccountOwner::User(owner) => owner,
-            AccountOwner::Application(_) => panic!("Applications not supported yet!"),
-        };
-        Ok(system_api::current_owner_balance(owner))
-    }
-
     async fn ticker_symbol(&self) -> Result<String, async_graphql::Error> {
         Ok(String::from(TICKER_SYMBOL))
     }
+
+    async fn accounts(&self) -> Result<MapView<AccountOwner, Amount>, async_graphql::Error> {
+        let mut accounts: MapView<AccountOwner, Amount> =
+            MapView::load(self.context.clone()).await?;
+        let balances = system_api::all_current_owner_balances();
+
+        balances.iter().for_each(|(owner, amount)| {
+            let account_owner = AccountOwner::User(*owner);
+            accounts
+                .insert(&account_owner, *amount)
+                .expect("Inserting into MapView should not fail");
+        });
+        Ok(accounts)
+    }
```

Comment:
It would be good to return a view that only loads the entries that are actually needed. (But maybe that's also a separate PR.)

Reply:
I see, but that would have to be a different query I think. The query now assumes it's getting a map back because it uses `entry`, for example.

Reply:
Yes, it would have to be a type that implements the same interface as a `MapView`, but is backed by different data.